src/Adapters/AwsS3Connector.php

```diff
@@ -68,9 +64,15 @@ protected function getAuth(array $config)
         $auth = [
             'region'      => $config['region'],
             'version'     => $config['version'],
-            'credentials' => array_only($config, ['key', 'secret']),
         ];
 
+        if (array_key_exists('key', $config) && $config['key'] != null) {
```

Comment:
I think `isset($config['key'])` is what we want here?

Reply:
yeah more succinct.